src/index.ts

```diff
@@ -1,54 +1,40 @@
-const path = require('path');
-const { RawSource } = require('webpack-sources');
-
-class MiniHtmlWebpackPlugin {
-	constructor(options = {}) {
-		this.options = options;
-		this.plugin = this.plugin.bind(this);
-	}
-
-	plugin(compilation, callback) {
-		const {
-			filename = 'index.html',
-			publicPath = '',
-			template,
-			context,
-			chunks,
-		} = this.options;
-
-		const files = getFiles(
-			normalizeEntrypoints(compilation.entrypoints),
-			chunks
-		);
-
-		const options = Object.assign({}, { publicPath }, context, files);
-
-		Promise.resolve((template || defaultTemplate)(options)).then(source => {
-			compilation.assets[filename] = new RawSource(source);
-			callback();
-		});
-	}
-
-	apply(compiler) {
-		if (compiler.hooks) {
-			// Webpack 4
-			compiler.hooks.emit.tapAsync('MiniHtmlWebpackPlugin', this.plugin);
-		} else {
-			// Webpack 3
-			compiler.plugin('emit', this.plugin);
-		}
-	}
-}
-
-function getFiles(entrypoints, chunks) {
-	const ret = {};
-
-	entrypoints.forEach(entry => {
+import path from 'path';
+import { RawSource } from 'webpack-sources';
+
+/* eslint-disable-next-line no-unused-vars */
+import webpack from 'webpack';
+
+type Context = {
+	title?: string;
+	htmlAttributes?: object;
```

Comment:
```suggestion
	htmlAttributes?: Record<string, any>;
```

would be a bit more precise.